### PR TITLE
Fix hexen coop demo playback

### PR DESCRIPTION
**Pull request by kraflab**
See https://github.com/chocolate-doom/chocolate-doom/pull/1358

cc @fabiangreffrath

### CODE REVIEW

**Review by fabiangreffrath — APPROVED** [review]

File: src/hexen/g_game.c [hunk 1]
```diff
@@ -2093,6 +2095,12 @@ void G_DoPlayDemo(void)
         PlayerClass[i] = *demo_p++;
     }
 
+    if (playeringame[1] || M_CheckParm("-solo-net") > 0
```
Comment (fragglet) [inline src/hexen/g_game.c:2098]: You can use M_ParmExists() for simplicity.

File: src/hexen/g_game.c [hunk 2]
```diff
@@ -2101,6 +2109,11 @@ void G_DoPlayDemo(void)
     precache = true;
     usergame = false;
     demoplayback = true;
+
+    if (netgame == true)
```
Comment (fragglet) [inline src/hexen/g_game.c:2113]: Please don't write `x == true`, it's the same as just `x`. Similarly `x == false` is just `!x`.

**Review by fragglet — CHANGES_REQUESTED** [review]
Thanks for doing this!

File: src/hexen/g_game.c (hunk 2 above)
Comment (kraflab) [reply src/hexen/g_game.c:2113]: I'm aware 😉 
I left it this way for consistency but I'm happy to change it.

File: src/hexen/g_game.c (hunk 1 above)
Comment (kraflab) [reply src/hexen/g_game.c]: Updated

File: src/hexen/g_game.c (hunk 2 above)
Comment (kraflab) [reply src/hexen/g_game.c]: Updated

**Review by fragglet — APPROVED** [review]

**Comment by fragglet** [comment]
Thanks.

**Pull request merged**